flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -862,11 +862,19 @@ private void onTriggerSuccess() {
      */
     private void onTriggerFailure(
             CheckpointTriggerRequest onCompletionPromise, Throwable throwable) {
-        final CheckpointException checkpointException =
-                getCheckpointException(
-                        CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE, throwable);
-        onCompletionPromise.completeExceptionally(checkpointException);
-        onTriggerFailure((PendingCheckpoint) null, checkpointException);
+        Throwable rootThrowable = ExceptionUtils.stripCompletionException(throwable);
```

Comment:
It is actually not a root throwable. stripCompletionException only unwraps CompletionException. But what if IOException would be hidden deeper than one level? According to the task description, I suppose we interesting in any IOException at any cause level. So if I understand everything right `ExceptionUtils.findThrowable(IOException).isPresent()` is more correct construction here.

Reply:
Also, why do you use unwrapped throwable only if it is IOException but if it is not you use the initial one? I think logic should be consistent here. If we know that this throwable can be wrapped by CompletionException we should unwrap it in both cases(Perhaps, it makes sense to move unwrapping into `getCheckpointException`).

Reply:
Sure, I have push a new commit and make the change.